File: src/main/scala/br/com/fabioaraujo/calcula/produto/dao/ItemRepository.scala
package br.com.fabioaraujo.calcula.produto.dao

import br.com.fabioaraujo.calcula.produto.dto.Item
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
trait ItemRepository extends CrudRepository[Item, java.lang.Long] {
  def findItemByNome(nome: String): Item
}

File: src/main/java/br/com/fabioaraujo/calcula/produto/dao/ItemRepository.java
package br.com.fabioaraujo.calcula.produto.dao;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import br.com.fabioaraujo.calcula.produto.dto.Item;

@Repository
public interface ItemRepository extends CrudRepository<Item, Long> {
	public Item findItemByNome(String nome);
}